server/src/main/java/org/spine3/server/entity/storage/EntityRecordWithStorageFields.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.entity.storage;
+
+import com.google.common.collect.ImmutableMap;
+import org.spine3.server.entity.EntityRecord;
+import org.spine3.server.entity.storage.reflect.Column;
+
+import javax.annotation.Nullable;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A packed value of an {@link EntityRecord} with its {@linkplain StorageFields storage fields}.
+ *
+ * @author Dmytro Dashenkov
+ */
+public class EntityRecordWithStorageFields {
+
+    private final EntityRecord record;
+
+    @Nullable
+    private final ImmutableMap<String, Column.MemoizedValue<?>> storageFields;
+
+    /**
+     * Creates a new instance of the {@code EntityRecordWithStorageFields}.
+     *
+     * @param record        {@link EntityRecord} to pack
+     * @param storageFields {@linkplain StorageFields storage fields map} to pack
+     */
+    public EntityRecordWithStorageFields(EntityRecord record,
```

Comment:
I'd make it `private` and create a factory method instead — to comply with the construction approach we have for value objects.